src/frontend/hooks/useWifiStatus.ts

```diff
@@ -0,0 +1,83 @@
+import * as React from "react";
+import NetInfo, { NetInfoSubscription } from "@react-native-community/netinfo";
+import KeepAwake from "react-native-keep-awake";
+import { NetworkInfo } from "react-native-network-info";
+
+import bugsnag from "../lib/logger";
+
+const useWifiStatus = ({
+  keepAwake,
+  onConnectionChange,
+  setListening,
+}: {
+  keepAwake?: boolean;
+  onConnectionChange?: () => void;
+  setListening?: (value: boolean) => void;
+} = {}) => {
+  const [ssid, setSsid] = React.useState<string | null>(null);
+
+  React.useEffect(() => {
+    const subscriptions: { remove: NetInfoSubscription }[] = [];
+
+    const handleConnectionChange = async () => {
+      // NetInfoData does not actually tell us whether wifi is turned on, it just
+      // tells us what connection the phone is using for data. E.g. it could be
+      // connected to a wifi network but instead using 4g for data, in which case
+      // `data.type` will not be wifi. So instead we just use the event listener
+      // from NetInfo, and when the connection changes we look up the SSID to see
+      // whether the user is connected to a wifi network.
+      // TODO: We currently do not know whether wifi is turned off, we only know
+      // whether the user is connected to a wifi network or not.
+      let ssid = null;
+      try {
+        ssid = await NetworkInfo.getSSID();
+      } catch (e) {
+        bugsnag.notify(e);
+      } finally {
+        // Even if we don't get the SSID, we still want to show that a wifi
+        // network is connected.
+        setSsid(ssid);
+
+        if (onConnectionChange) {
+          onConnectionChange();
+        }
+
+        if (setListening) {
+          setListening(true);
+        }
+      }
+    };
+
+    if (setListening) {
```

Comment:
This does not belong here, it can sit it its own useEffect in the sync screen.

Reply:
will pose this as a comment on the commit that updates this hook, but is the listening state directly related to the presence of `ssid` or should they be treated as different things? We use `listen` as the first argument of `usePeers` but based on your comment and just from a quick look at the code, it seems like `usePeers` could use the presence of `ssid` instead of a separate `listening` variable. Possible im missing something apparent

Reply:
Yes, I can't see any reason why this would not work. Looks good to me. This sync code needs a cleanup at some point! It's too complicated.